### PR TITLE
bsunanda:Run2-hcx120 Correct the bug for TestNumbering

### PR DESCRIPTION
**Pull request by bsunanda**
Also add a test for Relabelling
Fix issues for getting depth index from layer number in Plan-1 type scenario

### CODE REVIEW

**Comment by bsunanda** [comment]
@cmsbuild Please test

File: Geometry/HcalCommonData/src/HcalDDDRecConstants.cc [hunk 1]
```diff
@@ -185,7 +185,7 @@ HcalDDDRecConstants::getHCID(int subdet, int keta, int iphi, int lay,
     ++phi0;
     unit     = hcons.unitPhi(phibin[eta-1]);
     phi      = hcons.phiNumber(phi0,unit);
-    depth    = hcons.findDepth(subdet,eta,phi,zside,lay);
+    depth    = hcons.findDepth(subdet,eta,phi,zside,lay-1);
```
Comment (davidlange6) [inline Geometry/HcalCommonData/src/HcalDDDRecConstants.cc]: Hi @bsunanda -so to summarize the issue, HcalDDDRecConstants.cc expects layers starting from "1" and HcalDDDSimConstants expects layers starting from "0"?

Comment (bsunanda) [inline Geometry/HcalCommonData/src/HcalDDDRecConstants.cc]: We always use layer # going from 1 to ... But some mapping parameters are stored as 0,.... This was properly taken care of in the Sim step but missed out for Plan 1 operation for TestNumbering option.

**Comment by bsunanda** [comment]
@civanch Please approve this PR. Urgently needed for Plan1 simulation

**Comment by civanch** [comment]
@ianna , can you sign this PR?

File: Geometry/HcalCommonData/src/HcalDDDRecConstants.cc (hunk 1 above)
Comment (ianna) [inline Geometry/HcalCommonData/src/HcalDDDRecConstants.cc]: @bsunanda - please, add this to the PR description. Thanks.

**Comment by ianna** [comment]
@bsunanda - CMSSW release notes include the list of PRs which went into a release. Some meaningful  description would help here.

**Comment by ianna** [comment]
+1

**Comment by civanch** [comment]
+1
I would follow Ianna comment

**Comment by davidlange6** [comment]
hi @bsunanda all - are there some validation plots given this change?

**Comment by abdoulline** [comment]
Before/after: spectrum shape and number of entries (when comparing HEP17 in question to its siblinh HEM17 on the other side of HCAL):

hep17_spectrum_900

hep17_hem17_spectrum_pr 18111

**Comment by abdoulline** [comment]
NB: actually all the validation have been done in 900+PR#18111

Maria's results: 
http://dalfonso.web.cern.ch/dalfonso/Validation_PR18111.pdf

Fedrico's results:
plan1_fixtestnumscheme

**Comment by abdoulline** [comment]
@davidlange6  
with these results, hope you can merge it (and PR #18111 for 90X)

**Comment by davidlange6** [comment]
+1
Thanks @abdoulline

**Pull request merged**